src/Language/PureScript/Errors.hs

```diff
@@ -1353,12 +1353,29 @@ prettyPrintSingleError (PPEOptions codeColor full level showDocs relPath) e = fl
       line $ "Duplicate role declaration for " <> markCode (runProperName name) <> "."
 
     renderSimpleErrorMessage (CannotApplyExpressionOfTypeOnType tabs targ) =
-      paras
+      paras $
         [ "An expression of type:"
         , markCodeBox $ indent $ prettyType tabs
         , "cannot be applied to the type:"
         , markCodeBox $ indent $ prettyType targ
-        ]
+        ] <> comments
+      where
+      comments =
+        case tabs of
+          ForAll a i k t s _
+            | not $ hasVta tabs
+            , isMonoType t ->
```

Comment:
I've decided to only suggest adding a visible type abstraction when there's a single `ForAll` as it's irrefutable that this is what the user wants.

Reply:
One problem here though is the specialized syntax for data constructors and class declarations, so maybe just having a plain suggestion here would be better.